File: app/src/main/java/com/example/wechatimitation/Msg.java
package com.example.wechatimitation;

import android.annotation.SuppressLint;
import android.content.Context;
import android.widget.Toast;

import java.text.SimpleDateFormat;
import java.util.Date;

public class Msg {

    public static final int TYPE_RECEIVED = 0;
    public static final int TYPE_SENT = 1;
    @SuppressLint("SimpleDateFormat")
    private final SimpleDateFormat TIME_FORMAT = new SimpleDateFormat("YYYY-MM-dd HH:mm:ss");
    private String content;
    private int type;
    private String timeStr;

    public Msg(){}

    public Msg(String content, int type) {
        this.content = content;
        this.type = type;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public int getType() {
        return type;
    }

    public void setType(int type) {
        this.type = type;
    }

    public Date getTime() {
        try {
            return TIME_FORMAT.parse(timeStr);
        } catch (Exception e) {
            e.printStackTrace();
        }
        return new Date();
    }

    public String getTimeStr() {
        return timeStr;
    }

    public void setTimeStr(String timeStr) {
        this.timeStr = timeStr;
    }

    public static void showText(Context parent, String text) {
        Toast tt = Toast.makeText(parent, text, Toast.LENGTH_SHORT);
        tt.show();
        tt = null;
    }

}

File: app/src/main/java/com/example/wechatimitation/Msg.kt
package com.example.wechatimitation

import android.annotation.SuppressLint
import android.content.Context
import android.widget.Toast
import java.text.SimpleDateFormat
import java.util.*

class Msg {
    @SuppressLint("SimpleDateFormat")
    private val TIME_FORMAT = SimpleDateFormat("YYYY-MM-dd HH:mm:ss")
    private var content: String? = null
    private var type = 0
    private var timeStr: String? = null

    constructor() {}
    constructor(content: String?, type: Int) {
        this.content = content
        this.type = type
    }

    fun getContent(): String? {
        return content
    }

    fun setContent(content: String?) {
        this.content = content
    }

    fun getType(): Int {
        return type
    }

    fun setType(type: Int) {
        this.type = type
    }

    fun getTime(): Date {
        try {
            return TIME_FORMAT.parse(timeStr)
        } catch (e: Exception) {
            e.printStackTrace()
        }
        return Date()
    }

    fun getTimeStr(): String? {
        return timeStr
    }

    fun setTimeStr(timeStr: String?) {
        this.timeStr = timeStr
    }

    companion object {
        const val TYPE_RECEIVED = 0
        const val TYPE_SENT = 1
        fun showText(parent: Context?, text: String?) {
            var tt = Toast.makeText(parent, text, Toast.LENGTH_SHORT)
            tt!!.show()
            tt = null
        }
    }
}